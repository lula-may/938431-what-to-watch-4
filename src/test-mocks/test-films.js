const movieTitles = [
  `Jurassic Park`,
  `Gone With The Wind`,
  `The Godfather`,
  `Casablanca`,
  `A Streetcar Named Desire`,
  `Some Likes It Hot`,
];

const actors = [
  [`Sam Neill`, `Jeff GoldBlum`, `Laura Dern`],
  [`Tom Cruise`, `Rayan Gosling`],
  [`Anne Hathaway`, `Nicole Kidman`, `Keanu Reeves`],
  [`Will Smith`, `Bruce Willis`, `Mila Yovovich`],
  [`Sandra Bullock`, `Daniel Craig`],
  [`Robert De Niro`, `Vin Diesel`, `Charlize Theron`]
];

const MAX_RATING = 9.7;
const RATING_COUNT = 150;
const MAX_RELEASE_YEAR = 2000;

const directors = [`Stiven Spielberg`, `Mr. Good`, `Mr. Bad`, `Mrs. Ugly`, `Ms. Red`, `Mr. Black`];

const genres = [`Sci-Fi`, `War`, `Detective`, `Fantasy`, `Sport`, `Adventure`];

const testMovies = movieTitles.map((title, i) => {
  const rating = (MAX_RATING - i).toFixed(1);
  return {
    actors: actors[i],
    bigPoster: `bip-poster-${i}.jpg`,
    src: `video-${i}.mp3`,
    description: {
      paragraphs: [`bla-bla-bla`, `olya-lya-lya`, `wow-wow`]
    },
    director: directors[i],
    genre: genres[i],
    id: i,
    poster: `poster-${i}.jpg`,
    rating: {
      count: RATING_COUNT,
      score: rating,
    },
    releaseYear: MAX_RELEASE_YEAR - i,
    title,
  };
});


export {testMovies};

