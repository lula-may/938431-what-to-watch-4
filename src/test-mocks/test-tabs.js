const tabs = [
  {
    id: `aa`,
    type: `First`
  },
  {
    id: `bb`,
    type: `Second`,
  },
  {
    id: `cc`,
    type: `Third`,
  }
];

export {tabs};
