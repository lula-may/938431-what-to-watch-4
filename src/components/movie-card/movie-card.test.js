import React from "react";
import renderer from "react-test-renderer";
import MovieCard from "./movie-card.jsx";
import {testMovies} from "../../test-mocks/test-films";

const movie = testMovies[4];

it(`should correctly render MovieCardComponent`, () => {
  const tree = renderer.create(
      <MovieCard
        movie={movie}
        onMouseEnter={() => {}}
        onCardClick={() => {}}
        renderVideo={() => {}}
      />
  )
  .toJSON();

  expect(tree).toMatchSnapshot();
});
