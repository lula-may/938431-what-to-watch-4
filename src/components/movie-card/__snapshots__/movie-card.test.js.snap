// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render MovieCardComponent 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
    onClick={[Function]}
  >
    <img
      alt="A Streetcar Named Desire"
      height="175"
      src="poster-4.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      A Streetcar Named Desire
    </a>
  </h3>
</article>
`;
